test/rubygems/test_gem_remote_fetcher.rb

```diff
@@ -603,4 +588,10 @@ def assert_error(exception_class = Exception)
 
     assert got_exception, "Expected exception conforming to #{exception_class}"
   end
+
+  def fake_fetcher(url, data)
+    fetcher = Gem::FakeFetcher.new
+    fetcher.data[url] = data
+    Gem::RemoteFetcher.fetcher = fetcher
```

Comment:
This mutates global state. Do we need to save the previous value and set it back after calling this method?

Reply:
I think technically it's fine because `setup` always sets `fetcher` to nil. See https://github.com/ruby/rubygems/blob/79df6f8ae4524e822b086873a33bd0bd525fc78b/test/rubygems/test_gem_remote_fetcher.rb#L23

but I'll add an ensure anyway in case something gets changed in the future.